scripts/build-and-score-data.js

```diff
@@ -173,39 +193,48 @@ const buildAndScoreData = async () => {
     );
   }
 
+  console.log('📄️ Preparing data file');
+
+  const { libraries, ...rest } = latestData;
+
+  let fileContent;
+
   if (wantedPackageName) {
-    const { libraries, ...rest } = JSON.parse(fs.readFileSync(DATA_PATH));
-
-    return fs.writeFileSync(
-      DATA_PATH,
-      JSON.stringify(
-        {
-          libraries: libraries.map(library => {
-            if (library.npmPkg === wantedPackageName) {
-              return data.find(entry => entry.npmPkg === wantedPackageName);
-            }
-            return library;
-          }),
-          ...rest,
-        },
-        null,
-        2
-      )
+    fileContent = JSON.stringify(
+      {
+        libraries: libraries.map(library => {
+          if (library.npmPkg === wantedPackageName) {
+            return data.find(entry => entry.npmPkg === wantedPackageName);
+          }
+          return library;
+        }),
+        ...rest,
+      },
+      null,
+      2
     );
   } else {
-    return fs.writeFileSync(
-      DATA_PATH,
-      JSON.stringify(
-        {
-          libraries: data,
-          topics: topicCounts,
-          topicsList: Object.keys(topicCounts).sort(),
-        },
-        null,
-        2
-      )
+    const existingData = libraries.map(lib => lib.githubUrl);
+    const newData = data.map(lib => lib.githubUrl);
+    const missingData = existingData.filter(url => !newData.includes(url));
+
+    fileContent = JSON.stringify(
+      {
+        libraries: [...libraries.filter(lib => missingData.includes(lib.githubUrl)), ...data],
+        topics: topicCounts,
+        topicsList: Object.keys(topicCounts).sort(),
+      },
+      null,
+      2
     );
   }
+
+  await uploadToStore(fileContent);
+
+  console.log('🗑️️ Removing old data blob from the store');
+  await del(blobUrl);
```

Comment:
maybe let's leave these around and create another weekly github action to delete old blobs?